crates/cast/tests/data/contracts/no_casm/src/lib.cairo

```diff
@@ -1,7 +1,15 @@
+#[starknet::interface]
+trait Iminimal_contract<TContractState> {
+    fn empty(ref self: TContractState);
+}
+
 #[starknet::contract]
 mod minimal_contract {
     #[storage]
     struct Storage {}
-    #[external(v0)]
-    fn empty(ref self: ContractState) {}
+
+    #[abi(embed_v0)]
+    impl minimal_contractImpl of super::Iminimal_contract<ContractState> {
+        fn empty(ref self: ContractState) {}
+    }
 }
```

Comment:
Soo we do have to define traits for all methods, right?

Reply:
I think technically you could still use `#[external(v0)]` because I don't think it got removed entirely (actually I've left it in one or two places because cast tests weren't passing). But generally from what I understand, yes. ` #[abi(embed_v0)]` will only work in context of impl of `starknet::interface` so we have to define them.

Reply:
I think this comment proves that it works without an interface
https://github.com/foundry-rs/starknet-foundry/pull/1292#discussion_r1420088407

Reply:
Oh, so dispatchers would not have any methods, right?